### PR TITLE
EI 2028 - Added private endpoint resources

### PR DESCRIPTION
**Pull request by swiftnickmoj**
### Jira link (if applicable)

https://tools.hmcts.net/jira/browse/EI-2028

### Change description ###
Added private endpoint resources

### Checklist

- [ ] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)
- [ ] tests have been updated / new tests has been added (if needed)
- [ ] Does this PR introduce a breaking change

### CODE REVIEW

File: main.tf [hunk 1]
```diff
@@ -47,6 +47,56 @@ resource "azurerm_linux_function_app" "linux_function" {
...
+      private_connection_resource_id = azurerm_windows_function_app.windows_function.id
+
+      subresource_name = ["site"]
+      is_manual_connection = false
+    }    
+  }
+  
+# Integrate with VNet
+resource "azurerm_app_service_virtual_network_swift_connection" "private_function_vnet_link" {
+  app_service_id = var.function_app_name.id
```
Comment (varun-vijayagopal) [inline main.tf:86]: Should this be conditional as well ?

**Comment by varun-vijayagopal** [comment]
@swiftnickmoj pre commit checks are failing . Can u please fix it

File: main.tf [hunk 2]
```diff
@@ -47,6 +47,56 @@ resource "azurerm_linux_function_app" "linux_function" {
...
+      private_connection_resource_id = azurerm_linux_function_app.linux_function.id
+
+      subresource_name = ["site"]
+      is_manual_connection = false
+    }    
+  }
+
+  resource "azurerm_private_endpoint "windows_private_endpoint" {
+    count           = var.asp_os_type == "Windows" && var.app_service_plan_type == "PremiumV2" ? 1 : 0
+    name = var.private_endpoint
```
Comment (varun-vijayagopal) [inline main.tf:70]: Is this an existing variable ?

**Review by varun-vijayagopal — CHANGES_REQUESTED** [review]

File: main.tf (hunk 1 above)
Comment (swiftnickmoj) [reply main.tf:86]: Yes, will address as per your suggestion.  Thanks!

File: main.tf (hunk 2 above)
Comment (swiftnickmoj) [reply main.tf:71]: This is a new variable.  I don't see it pre-defined elsewhere in the code.

**Comment by swiftnickmoj** [comment]
Superseded by latest P/R

**Pull request closed without merging**